src/NuGet.Core/NuGet.Commands/Services/IPackageMetadataService.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Configuration;
+
+namespace NuGet.Commands.Experimental.Services
+{
+    public interface IPackageMetadataService
+    {
+        Task<PackageMetadataSourceResult> GetLatestMetadataAsync(string packageId, IEnumerable<PackageSource> packageSources, bool includePrerelease = false, CancellationToken cancellationToken = default);
```

Comment:
I just remembered that there's an example from outside the nuget team that would benefit from an API like this: `dotnet tool install`. It would be very convenient for them if this API handled package source mapping for them, so they don't have to, meaning the API should take in an `ISettings`.

Reply:
I agree. 
We can always iterate on this in a follow-up, but I think the PSM need is there and any future functionality we build should be PackageSourceMapping aware.